packages/back-end/src/services/experiments.ts

```diff
@@ -813,6 +816,115 @@ export function toSnapshotApiInterface(
+  organization: OrganizationInterface,
+  apiMetric: Partial<ApiMetric> & RequiredApiMetricFields,
+  dataSource: DataSourceInterface
+): Partial<MetricInterface> {
+  const metricDefaults = organization.settings?.metricDefaults;
+
+  const {
+    datasourceId,
+    name,
+    description,
+    type,
+    behavior,
+    sql,
+    mixpanel,
+    owner,
+    tags = [],
+    projects = [],
+  } = apiMetric;
+
+  let queryFormat: undefined | "builder" | "sql" = undefined;
+
+  const sqlBuilder = sql?.builder;
+  if (sqlBuilder) {
+    queryFormat = "builder";
+  } else if (sql) {
+    queryFormat = "sql";
+  }
+
+  const conditionsForDataSourceType = (): Condition[] => {
+    switch (dataSource.type) {
+      case "mixpanel":
+        return (mixpanel?.conditions || []).map(
+          ({ operator, property, value }) => ({
+            column: property,
+            operator: operator as Operator,
+            value: value,
+          })
+        );
+
+      default:
+        return (sqlBuilder?.conditions || []) as Condition[];
+    }
+  };
+
+  const metric: Omit<MetricInterface, "dateCreated" | "dateUpdated" | "id"> = {
+    datasource: datasourceId,
+    type: type as MetricType,
+    organization: organization.id,
+    name,
+    queryFormat,
+    description: description || "",
+    runStarted: null,
+    owner: owner || "",
+    tags,
+    projects,
+    inverse: behavior?.goal === "decrease",
+    ignoreNulls: false,
+    aggregation: sql?.userAggregationSQL || mixpanel?.userAggregation || "",
+    sql: sql?.userAggregationSQL || sql?.conversionSQL, // TODO: Is this correct?
+    cap: behavior?.cap ?? 0,
+    conditions: conditionsForDataSourceType(),
+
+    conversionDelayHours: behavior?.conversionWindowStart ?? 0,
+    conversionWindowHours: behavior?.conversionWindowEnd ?? 0,
+    loseRisk: behavior?.riskThresholdDanger ?? 0.0125,
+    winRisk: behavior?.riskThresholdSuccess ?? 0.0025,
+    column: sqlBuilder?.valueColumnName || mixpanel?.eventValue, // Is this correct?
+    maxPercentChange:
+      behavior?.maxPercentChange ?? metricDefaults?.maxPercentageChange ?? 0.5,
+    minPercentChange:
+      behavior?.minPercentChange ??
+      metricDefaults?.minPercentageChange ??
+      0.005,
+    minSampleSize:
+      behavior?.minSampleSize ?? metricDefaults?.minimumSampleSize ?? 150,
+    denominator: sql?.denominatorMetricId || "",
+    table: mixpanel?.eventName || sqlBuilder?.tableName || "",
+    status: "active",
+    timestampColumn: sqlBuilder?.timestampColumnName || "timestamp",
+    queries: [],
+    userIdTypes: sql?.identifierTypes || [],
+    userIdType: undefined,
+    anonymousIdColumn: "", // TODO: ??
+    userIdColumn: "", // TODO: Remove?
+    userIdColumns: (sqlBuilder?.identifierTypeColumns || []).reduce<
+      Record<string, string>
+    >((acc, { columnName, identifierType }) => {
+      acc[columnName] = identifierType;
+      return acc;
+    }, {}),
+  };
+
+  return metric;
+}
```

Comment:
Looking for feedback for this util, which is called from the `postMetric` handler.